src/tools/pub/cmd/pub-outdated.md

```diff
@@ -0,0 +1,240 @@
+1. **Test** your package to confirm that it still works as expected.
+
+You might still have [transitive dependencies][] that are out-of-date.
+If you want to determine the cause,
+try running [`pub deps`][] and searching the output for
+the name of the out-of-date package.
+
+
+## Example
+
+Here's an example of running `pub outdated` on
+an example that depends on old versions of two packages:
+`args` and `pedantic`.
+As the following screenshot shows,
+`pub outdated` colorizes the output by default
+when you run it on the command line.
+
+![screenshot of pub outdated output; visible as text later in "Output columns" section](images/pub-outdated.png)
+
+The preceding output shows that one regular dependency (`args`)
+and one [dev dependency][] (`pedantic`) are out-of-date.
+For details on what each column shows, see the
+[output columns](#output-columns) section of this page.
+
+As a rule, you update the **dependencies** and **dev_dependencies** sections
+of your `pubspec.yaml` file
+so that each package uses the versions in the **Resolvable** column.
+In this example, the resolvable version of `args` is **1.6.0**,
+and the resolvable version of `pedantic` is **1.9.0**.
+
+{{site.alert.tip}}
+  To see what changed in the new version of a package
+  that's published on [pub.dev,]({{site.pub}})
+  look at the changelog in the package page.
+  For example, you can look at the **Changelog** tabs on the pages for the
+  [`args`][] and [`pedantic`][] packages.
+{{site.alert.end}}
+
+To fix the first dependency (`args`),
+all that's needed is to run `pub upgrade`:
+
+```terminal
+$ pub upgrade
+Resolving dependencies... 
+> args 1.6.0 (was 1.5.0)
+  pedantic 1.8.0 (1.9.0 available)
+Changed 1 dependency!
+```
+
+To fix the second dependency (`pedantic`),
+you can change the pubspec's `pedantic` entry to use [caret syntax][] (`^`)
+and the number in the **Resolvable** column: **`^1.9.0`**.
+Here's the diff for `pubspec.yaml`:
+
+{% comment %}
+  [TODO: Improve the formatting of the following diff]
+{% endcomment %}
+
+```diff
+-  pedantic: 1.8.0
++  pedantic: ^1.9.0
+```
+
+{{site.alert.note}}
+  Updating the `pedantic` package can cause code to start failing analysis,
+  depending on your code and how you include the analysis options file.
+  For details, see the section on
+  [enabling default Google rules][using pedantic]
+  in [Customizing static analysis][].
+{{site.alert.end}}
+
+After editing `pubspec.yaml`, run `pub upgrade` to
+update the `pubspec.lock` file.
+You can then run `pub outdated` to confirm that
+you've made all necessary changes.
+
+```terminal
+$ pub upgrade
+...
+$ pub outdated
+Found no outdated packages.
+```
+
+{{site.alert.important}}
+  Test your code to verify that it still works as expected
+  after updating the packages.
+{{site.alert.end}}
+
+
+## Output columns
+
+The output of `pub outdated` has four columns of version information
+for each out-of-date dependency:
+
+```terminal
```

Comment:
Consider using a screenshot so we get the colors again

Reply:
The problem with a screenshot is that it's very bad for accessibility and search.

Reply:
That's a good point. Can we get colors (specifically the red highlight) in markdown?

Reply:
Use:
```bash
sudo apt-get install aha
TERM=xterm-256color script -qefc 'dart bin/pub.dart outdated' | aha -n
```

Then wrap the output with `<pre></pre>`. See `man aha` if you want colors for a black background, and other tweaks.

The `TERM=xterm-256color script -qefc 'dart ...'` wrapping just helps to trick dart:io into thinking it's running in a terminal with ANSI color support.

Reply:
Michael, can you do this? I don't have access to a Linux machine. (I was able to use brew to install aha, but the script command is different on Mac — doesn't let you specify a command, e.g. — and it'd take quite a while to figure this out.)

Reply:
https://gist.github.com/mit-mit/a20b65967d8cd0c6c33f93a8b0363a01

Reply:
Got it, thanks!